lib/ansiblelint/__main__.py

```diff
@@ -77,10 +76,8 @@ def main() -> int:
 
     formatter = formatter_factory(cwd, options.display_relative_path)
 
-    if options.use_default_rules:
-        rulesdirs = options.rulesdir + [DEFAULT_RULESDIR]
-    else:
-        rulesdirs = options.rulesdir or [DEFAULT_RULESDIR]
+    rulesdirs = get_rules_dirs([str(rdir) for rdir in options.rulesdir],
```

Comment:
We should operate pathlib objects internally, no strings until really necessary.

Reply:
I also prefer that but current code looks expecting rulesdirs as typing.List[str], e.g. ansiblelint.utils.expand_path{,s}_vars.

Reply:
Normally, str-conversions should be done when the code is passed to functions that expect strings but before that it'd use pathlib objects. Though, it's probably not a big deal atm.